cmd/internal/legacy/util.go

```diff
@@ -81,12 +94,13 @@ func startLearnSession(c rest.LearnClient, baseSpecRef *kgxapi.APISpecReference,
 	return lrn, nil
 }
 
-func checkpointLearnSession(c rest.LearnClient, lrn akid.LearnSessionID, timeout time.Duration) (akid.APISpecID, error) {
+func checkpointLearnSession(c rest.LearnClient, lrn akid.LearnSessionID, timeout time.Duration) (akid.APISpecID, string, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
-	specID, err := c.CheckpointLearnSession(ctx, lrn)
+	name := util.RandomAPIModelName()
+	specID, err := c.CreateSpec(ctx, name, []akid.LearnSessionID{lrn}, rest.CreateSpecOptions{})
```

Comment:
Hooray!

Reply:
KGX auto-generates a name when one isn't provided: https://github.com/akitasoftware/superstar/blob/2cb7c7b1239ea94a4a7ab8cde699a85c38014e2d/services/kings_cross/specs_db/pg.go#L90-L92. The response doesn't include this name, however.